telemetry/statsd/src/main/scala/io/buoyant/telemetry/StatsDInitializer.scala

```diff
@@ -0,0 +1,56 @@
+package io.buoyant.telemetry
+
+import com.timgroup.statsd.NonBlockingStatsDClient
+import com.twitter.finagle.Stack
+import com.twitter.finagle.util.DefaultTimer
+import com.twitter.logging.Logger
+import io.buoyant.telemetry.statsd.{StatsDStatsReceiver, StatsDTelemeter}
+import java.util.UUID
+
+class StatsDInitializer extends TelemeterInitializer {
+  type Config = StatsDConfig
+  val configClass = classOf[StatsDConfig]
+  override val configId = "io.l5d.statsd"
+}
+
+private[telemetry] object StatsDConfig {
+  val DefaultPrefix = "linkerd"
+  val DefaultHostname = "127.0.0.1"
+  val DefaultPort = 8125
+  val DefaultGaugePeriodMs = 10000
+
+  val MaxQueueSize = 10000
+  val ProcessId = "l5d-uuid-" + UUID.randomUUID().toString
+}
+
+case class StatsDConfig(
+  prefix: Option[String],
+  hostname: Option[String],
+  port: Option[Int],
+  gaugePeriodMs: Option[Int]
+) extends TelemeterConfig {
+  import StatsDConfig._
+
+  private[this] val log = Logger.get("io.l5d.statsd")
+
+  // prefix format is: "prefix_l5d-uuid-process-id"
+  val statsDPrefix = prefix.getOrElse(DefaultPrefix) + "_" + ProcessId
+  val statsDHost = hostname.getOrElse(DefaultHostname)
+  val statsDPort = port.getOrElse(DefaultPort)
+
+  // initiate a UDP connection at startup time
+  log.info(s"connecting to StatsD at $statsDHost:$statsDPort as $statsDPrefix")
+  private[this] val statsDClient = new NonBlockingStatsDClient(
```

Comment:
Suggestion: We Should use the [datadog Statsd client](https://github.com/DataDog/java-dogstatsd-client)?

> A statsd client library implemented in Java. Allows for Java applications to easily communicate with statsd.
> This version was originally forked from java-dogstatsd-client and java-statsd-client but it is now the canonical home for the java-dogstatsd-client. Collaborating with the former upstream projects we have now combined efforts to provide a single release.

[DataDog Statsd Client](https://github.com/DataDog/java-dogstatsd-client)
> https://github.com/DataDog/java-dogstatsd-client/blob/master/src/main/java/com/timgroup/statsd/NonBlockingStatsDClient.java

// We can then support Datadog tagging (well we are using Datadog) for free.

```
 private static final StatsDClient statsd = new NonBlockingStatsDClient(
   statsDPrefix,                       /* prefix to any stats; may be null or empty string */
   statsDHost,                        /* common case: localhost */
   statsDPort,                         /* port */
    MaxQueueSize,                   /*queuesize*/
   listofTagFromConfig         // String[]    
            /* Datadog extension: Constant tags, always applied, this can be passed down from config optionally*/
  );
 
```